### PR TITLE
Page: More easily support full height pages

### PR DESCRIPTION
**Pull request by torkelo**
In https://github.com/grafana/grafana/pull/56278 I removed display: 'flex' from the pageInner div of the Page component as children of Page that had flexGrow (like Alert) did take up the remaining space on some pages. 

Because pageInner also has the page header, and then children for children to use height 100% (of the remaining height subtracting the header) pageInner really needs to be a flex container. 

Think one way to solve this is to add a wrapping div around children instead. This way a child of Page can use css height 100% safely (and get the remaining height left over). 

It's a bit hard to see in the diff, but he new markup is like this:

```tsx
        <div className={styles.pageContainer}>
            <CustomScrollbar autoHeightMin={'100%'} scrollTop={scrollTop} scrollRefCallback={scrollRef}>
              <div className={styles.pageInner}>
                {pageHeaderNav && <PageHeader navItem={pageHeaderNav} subTitle={subTitle} />}
                {pageNav && pageNav.children && <PageTabs navItem={pageNav} />}
                <div className={styles.pageContent}>{children}</div>
              </div>
              <Footer />
            </CustomScrollbar>
          </div>
        </div>
```

Basically the only change is the new addion of the pageContent div wrapping children. I also took the opertunity to rename / change places for the class names so the make better sense (pageContainer used to be named pageContent)

The other option is to pass in a custom pageInnerClass and mix the two but that feels more complex and harder to maintain. But it is close.

### CODE REVIEW

File: public/app/core/components/PageNew/Page.tsx [hunk 1]
```diff
@@ -167,6 +171,8 @@ const getStyles = (theme: GrafanaTheme2) => {
       boxShadow: shadow,
       background: theme.colors.background.primary,
       margin: theme.spacing(2, 2, 2, 1),
+      display: 'flex',
+      flexDirection: 'column',
```
Comment (torkelo) [inline public/app/core/components/PageNew/Page.tsx:175]: this is adding these back to pageInner (but this does not affect Page children as pageContent is now the wrapping class for children and it is a display block object with flexGrow: 1)

**Review by kaydelaney — APPROVED** [review]

**Review by ashharrison90 — APPROVED** [review]
lgtm 👍
